portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -31,20 +33,58 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+import java.text.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/delete-data")
 public class DeleteDataServlet extends HttpServlet {
-
+  
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").setKeysOnly();
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);    
+    PreparedQuery results = datastore.prepare(query);
+    String stringPostTime = (String) request.getParameter("postTime");
+    DateFormat format = new SimpleDateFormat("MMM d, yyyy h:mm:ss a");
+    Date postTime;
+    try {
+      postTime = format.parse(stringPostTime);
+    } catch (Exception e) {
+      System.out.println(e);
+      postTime = new Date();
+    }
+    UserService userService = UserServiceFactory.getUserService();
+    String userId = userService.getCurrentUser().getUserId();
+    String isAdmin = String.valueOf(userService.isUserAdmin());
     for (Entity entity : results.asIterable()) {
-      Key key = entity.getKey();
-      datastore.delete(key);
+      Date e_postTime = (Date) entity.getProperty("postTime");
+      String e_userId = (String) entity.getProperty("userId");
+      // Remove miliseconds from e_postTime
+      long time = e_postTime.getTime();
+      e_postTime.setTime((time / 1000) * 1000);
+      // Assume it's the right comment if it was posted at the same second
+      if (e_postTime.equals(postTime) && (e_userId.equals(userId) || isAdmin.equals("true"))){
+        Key key = entity.getKey();
+        datastore.delete(key);
+        System.out.println("deleted a comment");
```

Comment:
Could also be useful to return this status back to the client, for example so client can remove the comment on the screen without refreshing the page.